### ISSUE TITLE
Missing input validation for PReLU alpha_initializer and LeakyReLU alpha

### ISSUE THREAD

**Issue by amadhan882** [keras-team/keras#22162]
### Summary
`keras.layers.PReLU` and `keras.layers.LeakyReLU` lack sufficient input validation during instantiation and the `build()` phase. 

1. **PReLU:** Accepts `alpha_initializer=None`, leading to an inconsistent state that results in downstream failures during training or serialization.
2. **LeakyReLU:** Accepts `NaN` values for the `alpha` parameter, allowing the creation of a broken layer that produces non-numeric gradients.

This is a **usability and reliability bug** that violates the "Fail-Fast" principle of the Keras API.

### Reproduction Script
```python
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
import keras
import numpy as np

print(f"Testing Keras version: {keras.__version__}")

# --- Case 1: PReLU ---
print("\n--- Testing PReLU ---")
try:
    layer_prelu = keras.layers.PReLU(alpha_initializer=None)
    layer_prelu.build((None, 10)) 
    print("[ISSUE] PReLU accepted None as alpha_initializer and built successfully!")
except Exception as e:
    print(f"[OK] PReLU caught error: {e}")

# --- Case 2: LeakyReLU ---
print("\n--- Testing LeakyReLU ---")
try:
    layer_leaky = keras.layers.LeakyReLU(alpha=float('nan'))
    print("[ISSUE] LeakyReLU accepted NaN as alpha!")
except Exception as e:
    print(f"[OK] LeakyReLU caught NaN.")
```

### Observed Result
```text
Testing Keras version: 3.14.0
--- Testing PReLU ---
[ISSUE] PReLU accepted None as alpha_initializer and built successfully!

--- Testing LeakyReLU ---
[ISSUE] LeakyReLU accepted NaN as alpha!
```

### Actual Behavior
The layers are instantiated and built without any validation errors. The `PReLU` layer remains in an inconsistent state, and `LeakyReLU` accepts mathematically undefined values (`NaN`), deferring crashes to the training phase.

### Expected Behavior
The constructor should raise a `ValueError` immediately if:
- `alpha_initializer` is `None` in `PReLU`.
- The `alpha` (or `negative_slope`) parameter in `LeakyReLU` is `NaN`.

### Root Cause
In `keras/src/layers/activations/prelu.py` and `leaky_relu.py`, parameters are assigned to the object's state without being passed through a validation utility. This creates a "silent failure" scenario where architecture errors are caught too late in the ML lifecycle.

### Suggested Fix

**For PReLU:**
```python
if alpha_initializer is None:
    raise ValueError("The `alpha_initializer` argument for PReLU cannot be None.")
```

**For LeakyReLU:**
```python
if np.isnan(alpha):
    raise ValueError("The `alpha/negative_slope` argument for LeakyReLU cannot be NaN.")
```

### Environment
* **OS:** Windows 11
* **Python:** 3.13.5
* **Keras:** 3.14.0
* **Backend:** TensorFlow 2.20.0

### Conclusion
Implementing strict input validation ensures that architecture inconsistencies are caught during the model definition phase, preventing silent failures and wasted computational resources during training execution.

**Comment by adamsaidane** [comment]
Hello
I’d like to work on this issue and submit a PR to add the appropriate input validation and tests.

**Comment by amadhan882** [comment]
Hi @adamsaidane and @dhantule,

Thanks for confirming the issue and for the quick PR #22163.

It's great to see the "Fail-Fast" principle being applied here. Catching None initializers and NaN values during the instantiation/build phase will definitely prevent a lot of silent failures and debugging headaches for users during training.

Looking forward to seeing this merged!
